hpx/runtime/threads/policies/local_workstealing_scheduler.hpp

```diff
@@ -0,0 +1,1608 @@
+            std::size_t num_thread = std::size_t(-1)) const
+        {
+            // Return average task wait time of one specific queue.
+            // Return average thread wait time of one specific queue.
+            std::uint64_t wait_time = 0;
+            std::uint64_t count = 0;
+            if (std::size_t(-1) != num_thread)
+            {
+                HPX_ASSERT(num_thread < num_queues_);
+
+                auto const& d = data_[num_thread].data_;
+                if (num_thread < num_high_priority_queues_)
+                {
+                    wait_time =
+                        d.high_priority_queue_->get_average_task_wait_time();
+                    ++count;
+                }
+
+                if (num_thread == num_queues_ - 1)
+                {
+                    wait_time +=
+                        low_priority_queue_.get_average_task_wait_time();
+                    ++count;
+                }
+
+                wait_time += d.queue_->get_average_task_wait_time();
+                return wait_time / (count + 1);
+            }
+
+            for (std::size_t i = 0; i != num_queues_; ++i)
+            {
+                auto const& d = data_[num_thread].data_;
+                if (num_thread < num_high_priority_queues_)
+                {
+                    wait_time +=
+                        d.high_priority_queue_->get_average_task_wait_time();
+                }
+                wait_time += d.queue_->get_average_task_wait_time();
+                ++count;
+            }
+
+            return (wait_time +
+                       low_priority_queue_.get_average_task_wait_time()) /
+                (count + 1);
+        }
+#endif
+
+        // return a random victim for the current stealing operation
+        std::size_t random_victim(steal_request const& req) noexcept
+        {
+            std::size_t bucket_size =
+                (std::numeric_limits<std::int16_t>::max)() / num_queues_;
+
+            // generate 3 random numbers max before resorting to more expensive
+            // algorithm
+            std::size_t result = 0;
+            int attempts = 0;
+            do
+            {
+                result = uniform_int_(gen_) / bucket_size;
+                if (result < num_queues_ && result != req.num_thread_ &&
+                    !test(req.victims_, result))
+                {
+                    return result;
+                }
+            } while (++attempts < 3);
+
+            // to avoid infinite trials we randomly select one of the possible
+            // victims
+            std::size_t const num_victims = num_queues_ - count(req.victims_);
+
+            // generate one more random number
+            std::size_t selected_victim = 0;
+            bucket_size =
+                (std::numeric_limits<std::int16_t>::max)() / num_victims;
+            do
+            {
+                selected_victim = uniform_int_(gen_) / bucket_size;
```

Comment:
Could the random numbers be drawn directly from the correct range with `uniform_int_distribution(0, num_victims)`? Same in the previous while-loop.